metrics_sdk/lib/opentelemetry/sdk/metrics/aggregation/explicit_bucket_histogram.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Aggregation
+        # Contains the implementation of the ExplicitBucketHistogram aggregation
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation
+        class ExplicitBucketHistogram
+          DEFAULT_BOUNDARIES = [0, 5, 10, 25, 50, 75, 100, 250, 500, 1000].freeze
+          private_constant :DEFAULT_BOUNDARIES
+
+          # The default value for boundaries represents the following buckets:
+          # (-inf, 0], (0, 5.0], (5.0, 10.0], (10.0, 25.0], (25.0, 50.0],
+          # (50.0, 75.0], (75.0, 100.0], (100.0, 250.0], (250.0, 500.0],
+          # (500.0, 1000.0], (1000.0, +inf)
+          def initialize(
+            boundaries: DEFAULT_BOUNDARIES,
+            record_min_max: true
+          )
+            @data_points = {}
+            @aggregation_temporality = :delta
+            @boundaries = boundaries && !boundaries.empty? ? boundaries.sort : nil
+            @record_min_max = record_min_max
+          end
+
+          def collect(start_time, end_time)
+            @data_points.each_value do |hdp|
+              hdp.count = hdp.bucket_counts.sum if @boundaries
+              hdp.start_time_unix_nano = start_time
+              hdp.time_unix_nano = end_time
+            end
+            hdps = @data_points.values
+            if @aggregation_temporality == :delta
+              @data_points.clear
+            else
+              hdps = hdps.map do |hdp|
+                hdp = hdp.dup
+                hdp.bucket_counts = hdp.bucket_counts.dup
+                hdp
+              end
+            end
+            hdps
```

Comment:
This has the same issue with start time in non-`delta` as the `sum` aggregation. I think the same approach would be good here, distinguishing the two cases up front.

```suggestion
            if @aggregation_temporality == :delta
              # Set timestamps and 'move' data point values to result.
              hdps = @data_points.values.each do |hdp|
                hdp.start_time_unix_nano = start_time
                hdp.time_unix_nano = end_time
              end
              @data_points.clear
              hdps
            else
              # Update timestamps and take a snapshot.
              @data_points.values.map! do |hdp|
                hdp.start_time_unix_nano ||= start_time # Start time of a data point is from the first observation.
                hdp.time_unix_nano = end_time
                hdp = hdp.dup
                hdp.bucket_counts = hdp.bucket_counts.dup
                hdp
              end
            end
```

Note that this suggestion assumes we increment `hdp.count` in `update`, as suggested below.

Sidenote: there's an alternative approach where we extract the 4 cases into separate classes: `delta` or not, and buckets or not. Probably not worth doing that yet, but it's something to consider for later.